### PR TITLE
Scheduler - Fix pushing new appointments to the store (remoteFiltering: true) (T900529)

### PR DESCRIPTION
**Pull request by AntonSermyazhko**

### CODE REVIEW

**Review by andrewmakarov — DISMISSED** [review]

File: js/ui/scheduler/ui.scheduler.appointment_model.js [hunk 1]
```diff
@@ -323,17 +323,29 @@ class AppointmentModel {
     }
 
     _initStoreChangeHandlers() {
-        this._dataSource && this._dataSource.store()
-            .on('updating', ((newItem) => {
+        const dataSource = this._dataSource;
+        const store = dataSource && dataSource.store();
```
Comment (AryamnovEugeniy) [inline js/ui/scheduler/ui.scheduler.appointment_model.js]: I'd rather use `dataSource?.store()`

File: js/ui/scheduler/ui.scheduler.appointment_model.js [hunk 2]
```diff
@@ -323,17 +323,29 @@ class AppointmentModel {
...
-                }).bind(this));
-            }).bind(this));
+            });
+
+            store.on('push', pushItems => {
+                const items = dataSource.items();
+                const keyName = store.key();
+
+                pushItems.forEach(pushItem => {
+                    const isItemExists = items.filter(item => item[keyName] === pushItem.key).length !== 0;
```
Comment (AryamnovEugeniy) [inline js/ui/scheduler/ui.scheduler.appointment_model.js]: Sounds odd. What about `itemExists`?

**Review by AryamnovEugeniy — DISMISSED** [review]

File: testing/tests/DevExpress.ui.widgets.scheduler/common.tests.js [hunk 3]
```diff
@@ -637,6 +637,40 @@ QUnit.module('Initialization', {
...
+            views: ['week'],
+            currentView: 'week',
+            currentDate: new Date(2017, 4, 25)
+        });
+
+        const dataSource = this.instance.getDataSource();
+        dataSource.store().push([{ type: 'update', key: pushItem.id, data: pushItem }]);
+        dataSource.load();
+
+        const appointment = this.instance.$element().find('.dx-scheduler-appointment-title');
```
Comment (Yarovaya) [inline testing/tests/DevExpress.ui.widgets.scheduler/common.tests.js]: Can we use SchedulerTestWrapper in this test?

File: js/ui/scheduler/ui.scheduler.appointment_model.js (hunk 2 above)
Comment (AntonSermyazhko) [reply js/ui/scheduler/ui.scheduler.appointment_model.js]: Sure, it is old habit )

**Review by AryamnovEugeniy — DISMISSED** [review]

**Review by andrewmakarov — DISMISSED** [review]

**Review by andrewmakarov — APPROVED** [review]

**Review by AryamnovEugeniy — APPROVED** [review]

**Review by Yarovaya — APPROVED** [review]

**Pull request merged**